### PR TITLE
Issue 2169 - Node registration with special unicode character

### PR DESCRIPTION
**Pull request by sirotsinskuy**
Signed-off-by: Antonii Sirotsinskyi <asiro@softserveinc.com>

### CODE REVIEW

File: cli/register/register.go [hunk 1]
```diff
@@ -629,6 +629,17 @@ func verifyRegisterParamters(org, pattern, nodeOrgFromFlag string, patternFromFl
 		}
 	}
 
+	if nodeId, _ := cliutils.SplitIdToken(nodeIdTok); nodeId != "" {
+		illegalInput, err := api.InputIsIllegal(nodeId)
+		if err != nil {
+			cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, msgPrinter.Sprintf("could not validate node ID '%s': %v", nodeId, err))
+		} else if illegalInput != "" {
+			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("invalid node ID: A-Za-z0-9 and special symbols !-*+()?.,:&@ are only allowed"))
```
Comment (linggao) [inline cli/register/register.go:637]: "invalid node ID: A-Za-z0-9, unicode characters and special symbols !-*+()?.,:&@ are only allowed"

File: cli/register/register.go [hunk 2]
```diff
@@ -629,6 +629,17 @@ func verifyRegisterParamters(org, pattern, nodeOrgFromFlag string, patternFromFl
 		}
 	}
 
+	if nodeId, _ := cliutils.SplitIdToken(nodeIdTok); nodeId != "" {
```
Comment (linggao) [inline cli/register/register.go:632]: Should also consider getting node id from the agent if the nodeId is empty. See line 129-135. Agent gets the node id from file /etc/default/horizon. User could put a bad id there.

**Review by linggao — CHANGES_REQUESTED** [review]
per comments

File: cli/register/register.go (hunk 2 above)
Comment (sirotsinskuy) [reply cli/register/register.go]: Added `nodeId` fetching from agent in case of empty `nodeId`

**Review by linggao — APPROVED** [review]

**Pull request merged**